core/variables.tf

```diff
@@ -41,3 +71,40 @@ variable "rds_password" {
   type        = string
   sensitive   = true
 }
+
+variable "bastion_instance_type" {
+  description = "The bastion instance type"
+  default     = "t3.nano"
+}
+
+variable "bastion_instance_desired_count" {
+  description = "The desired number of the bastion instance"
+  default     = 1
+}
+
+variable "bastion_max_instance_count" {
+  description = "The maximum number of the instance"
+  default     = 1
+}
+
+variable "bastion_min_instance_count" {
+  description = "The minimum number of the instance"
+  default     = 1
+}
+
+variable "bastion_ami_id" {
+  description = "The AMI ID for the instance"
+  default     = "ami-01811d4912b4ccb26"
```

Comment:
image id can be fetched dynamically, that way you always will have latest version of OS for bastion

```hcl
data "aws_ami" "ecs_optimized" {
  most_recent = true
  owners      = ["amazon"]

  filter {
    name   = "virtualization-type"
    values = ["hvm"]
  }

  filter {
    name   = "name"
    values = ["amzn2-ami-ecs-hvm-*-x86_64-ebs"]
  }
}
```

And it can be used as `image_id = data.aws_ami.ecs_optimized.id`

Reply:
Nice one, updated in [c646df9](https://github.com/nvminhtue/tyson_devops_ic/pull/33/commits/c646df9c7e4137056eaf649e73847498c7450a32)
![image](https://github.com/user-attachments/assets/9d779ffd-25a1-454f-8fb9-dbc80577d0d9)